osu.Game/Screens/Select/DifficultyRecommender.cs

```diff
@@ -32,8 +32,6 @@ public class DifficultyRecommender : Component, IOnlineComponent
         private void load()
         {
             api.Register(this);
-
-            ruleset.ValueChanged += _ => bestRulesetOrder = null;
```

Comment:
``bestRulesetOrder`` should instead be reset when new values are written to ``recommendedStarDifficulty`` when the request succeeds.